url7f15a160d2_timofei7_old_ai_assignments_tgz-pJ8-org_json_TestJ8/src/hw5/P3bMain.java

```diff
@@ -0,0 +1,233 @@
+package hw5;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * this is CS44 W10 HW5 part 3b. 
+ * 
+ * @author tim
+ *
+ */
+public class P3bMain
+{
+
+	//private static final String trainingFile = "hw5/spellcheck_first.txt";  //gulliverstravels
+	//private static final String errorFile = "hw5/spellcheck_second_10perror.txt";  //gulliverstravels
+	private static final String trainingFile = "hw5/sptrain.txt"; 
+	private static final String errorFile = "hw5/sptrain_10perror.txt"; 
+
+	
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args)
+	{
+		
+		Scanner txtinput = new Scanner(System.in);
+		
+		while (true)
+		{
+			System.out.print("Pick option: prepare test files=1, run spellcheck on default file set=2:");
+			int option = txtinput.nextInt();
+			
+			switch(option)
+			{
+				case 1:
+					makefiles(txtinput);
+					break;
+				case 2:
+					spellcheck(txtinput);
+					break;
+				default:
+					System.out.println("I don't understand.. try again");
+			}
+			
+		}
+
+	}
+	
+	
+	/**
+	 * preps a file with errors
+	 * @param input
+	 */
+	private static void makefiles(Scanner input)
+	{
+		String file = "";
+		int errors = 0;
+		while (file=="")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.